src/main/java/com/adobe/target/edge/client/http/DefaultTargetHttpClient.java

```diff
@@ -62,6 +64,13 @@ public DefaultTargetHttpClient(ClientConfig clientConfig) {
       unirestInstance.config().addInterceptor(clientConfig.getRequestInterceptor());
     }
 
+    if (clientConfig.getHttpClient() != null) {
+      Config httpClientConfig = new Config();
```

Comment:
I think instead of creating a new (empty) Config() object, we should reuse the config instance that we're building when we call `unirestInstance.config()`.  And that would mean we should move your new code snippet all the way to the bottom of the DefaultTargetHttpClient constructor.  After making that change please re-run your manual test case to make sure it works as expected.